src/services/codefixes/fixAddMissingParam.ts

```diff
@@ -0,0 +1,248 @@
+    if (
+        signature &&
+        signature.declaration &&
+        isConvertableSignatureDeclaration(signature.declaration)
+    ) {
+        const declaration = (signature.declaration.body === undefined ? find(signature.declaration.symbol.declarations, d => isConvertableSignatureDeclaration(d) && !!d.body) : signature.declaration) as ConvertableSignatureDeclaration;
+        if (declaration === undefined) {
+            return undefined;
+        }
+
+        isSourceFileFromLibrary(program, declaration.getSourceFile());
+
+        const overload = signature.declaration.body === undefined ? signature.declaration : undefined;
+        if (overload && length(overload.parameters) !== length(declaration.parameters)) return undefined;
+
+        const name = tryGetName(declaration);
+        if (name === undefined) return undefined;
+
+        const declarations: ConvertableSignatureDeclaration[] = append([declaration], overload);
+        const newParameters: ParameterInfo[] = [];
+        const parametersLength = length(declaration.parameters);
+        const argumentsLength = length(callExpression.arguments);
+        if (parametersLength > argumentsLength) return undefined;
+
+        for (let i = 0, pos = 0, paramIndex = 0; i < argumentsLength; i++) {
+            const arg = callExpression.arguments[i];
+            const expr = isAccessExpression(arg) ? getNameOfAccessExpression(arg) : arg;
+            const type = checker.getWidenedType(checker.getBaseTypeOfLiteralType(checker.getTypeAtLocation(arg)));
+            const parameter = pos < parametersLength ? declaration.parameters[pos] : undefined;
+            if (
+                parameter &&
+                checker.isTypeAssignableTo(type, checker.getTypeAtLocation(parameter))
+            ) {
+                pos++;
+            }
+            else {
+                const newParameter = {
+                    pos: i,
+                    declaration: factory.createParameterDeclaration(
+                        /*modifiers*/ undefined,
+                        /*dotDotDotToken*/ undefined,
+                        expr && isIdentifier(expr) ? expr.text : `p${paramIndex++}`,
+                        /*questionToken*/ undefined,
+                        typeToTypeNode(checker, type, declaration),
+                        /*initializer*/ undefined,
+                    ),
+                };
+                append(newParameters, newParameter);
+            }
+        }
+        return {
+            declarations,
+            newParameters,
+            overload,
+            description: [
+                length(newParameters) > 1 ? Diagnostics.Add_missing_parameters_to_0 : Diagnostics.Add_missing_parameter_to_0,
+                declarationNameToString(name),
+            ],
+        };
+    }
+    return undefined;
+}
+
+function tryGetName(node: FunctionLikeDeclaration) {

```

Comment:
I vaguely remember something like this already existing as a utility. Something like getAssignedName maybe?

Reply:
`getAssignedName` doesn't handle `parameters` and `property declarations`.